backend/scripts/transfer-contract-ownership.ts

```diff
@@ -1,37 +1,117 @@
+  slugs: string[]
+  toUserId: string
+}
+
+interface User {
+  id: string
+  username: string
+  name: string
+  created_time: string
+}
+
+interface Contract {
+  id: string
+  data: any
+}
+
+const supabaseDirectClient = createSupabaseDirectClient()
+
+const argv = yargs(hideBin(process.argv))
+  .option('slugs', {
+    alias: 's',
+    type: 'array',
+    description: 'List of slugs to transfer',
+    demandOption: true,
+  })
+  .option('toUserId', {
+    alias: 'u',
+    type: 'string',
+    description: 'User ID to transfer contracts to',
+    demandOption: true,
+  })
+  .parseSync() as Args
+
+const BATCH_SIZE = 20
 
 if (require.main === module) {
-  runScript(async ({ pg, firestore }) => {
-    const fromUserId = 'zgCIqq8AmRUYVu6AdQ9vVEJN8On1'
-    const contractIdsToTransfer = await pg.map(
-      `select id from contracts where creator_id = $1`,
-      [fromUserId],
-      (row) => row.id
-    )
-    const toUserId = '4juQfJkFnwX9nws3dFOpz4gc1mi2'
-    const toUser = await pg.one(
-      `select * from users where id = $1`,
-      [toUserId],
-      (row) => convertUser(row)
-    )
-    console.log(
-      `Transferring ${contractIdsToTransfer.length} contracts from ${fromUserId} to ${toUser.name}`
-    )
-    const writer = new SafeBulkWriter()
-    for (const id of contractIdsToTransfer) {
-      const doc = firestore.collection('contracts').doc(id)
-      writer.update(doc, {
-        creatorId: toUser.id,
-        creatorAvatarUrl: toUser.avatarUrl,
-        creatorUsername: toUser.username,
-        creatorName: toUser.name,
-        creatorCreatedTime: toUser.createdTime,
-      })
-    }
-    await writer.close()
+  runScript(async () => {
+    const { slugs: slugsToTransfer, toUserId } = argv
+
+    try {
+      const toUserData = await supabaseDirectClient.oneOrNone<User>(
+        'SELECT id, username, name, created_time FROM users WHERE id = $1',
+        [toUserId]
+      )
 
-    console.log('done.')
+      if (!toUserData) {
+        console.error('User not found')
+        return
+      }
+
+      const toUser: User = toUserData
+
+      const contractsData = await supabaseDirectClient.any<Contract>(
+        'SELECT id, data FROM contracts WHERE slug IN ($1:csv)',
+        [slugsToTransfer]
+      )
+
+      if (contractsData.length === 0) {
+        console.error('No contracts found')
+        return
+      }
+
+      const contractsToTransfer: Contract[] = contractsData
+
+      console.log(
+        `Transferring ${contractsToTransfer.length} contracts to ${toUser.name}`
+      )
+
+      for (let i = 0; i < contractsToTransfer.length; i += BATCH_SIZE) {
+        const batch = contractsToTransfer.slice(i, i + BATCH_SIZE)
+
+        const updates = batch.map(({ id, data }) => {
+          const description = data.description || ''
+          const creatorUsername = data.creatorUsername || 'Unknown creator'
+          const creatorName = data.creatorName || 'Unknown name'
+
+          const updatedDescription = `${
```

Comment:
This is not going to work with tiptap. If you really want to do this you have to inspect a tiptap json object and add a new object into the data that is of the type and value you want, i.e. paragraph.